### PR TITLE
Improve type definitions

### PR DESCRIPTION
**Pull request by swsnr**
This pull request addresses a few issues I stumbled upon while trying to move from my hand-written types to the types in this repo.    It's by no means complete (there were more issues), but I'm stuck in the process due to some other issues (see #10), so I figured I'd open a merge request with what I already got.

Recommended review is commit by commit, as every commit addresses one problem.

### CODE REVIEW

**Comment by swsnr** [comment]
I've turned this into a draft as now that I've overcome #10 I'll a few more fixes ready.

**Comment by swsnr** [comment]
I think I've fixed all errors I've hit in my extension, see https://github.com/swsnr/gnome-shell-extension-picture-of-the-day/pull/38

File: packages/gnome-shell/src/types/extension-metadata.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-    "shell-version"?: string[];
+    readonly "shell-version": string[];
     "settings-schema"?: string;
     "gettext-domain"?: string;
     "original-author"?: string[];
     "extension-id"?: string;
-}
\ No newline at end of file
+    "version-name"?: string;
+}
```
Comment (schnz) [inline packages/gnome-shell/src/types/extension-metadata.ts:17]: No reason to withheld the `readonly` modifier from the optional fields comes to mind. Have you withheld it on purpose from those fields?

File: packages/gnome-shell/src/ui/main.d.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ import { AudioDeviceSelectionDBus } from './audioDeviceSelection.js';
 // import * as InputMethod from '../misc/inputMethod.js'
 // import * as Introspect from '../misc/introspect.js'
 // import * as Keyboard from './keyboard.js';
-// import * as MessageTray from './messageTray.js';
+import * as MessageTray from './messageTray.js';
```
Comment (schnz) [inline packages/gnome-shell/src/ui/main.d.ts:20]: I feel bad for nitpicking here, but can you change the import to this (for consistency reasons)?

```suggestion
import { MessageTray } from './messageTray.js';
```

**Review by schnz — COMMENTED** [review]
Nice! Looks good to me! Maybe you can comment on the `readonly` comment below and then it's ready to be merged :)

---

… and a happy new year 🥂!

File: packages/gnome-shell/src/types/extension-metadata.ts (hunk 1 above)
Comment (swsnr) [reply packages/gnome-shell/src/types/extension-metadata.ts:17]: No particular reason; I just blindly made all properties readonly as documented in the guide and didn't care about the rest 😇 

I've made them all readonly.

File: packages/gnome-shell/src/ui/main.d.ts (hunk 2 above)
Comment (swsnr) [reply packages/gnome-shell/src/ui/main.d.ts]: Good point. done.

**Comment by swsnr** [comment]
@schnz Happy new year as well 🎆 

I've addressed your remarks 🙂

**Comment by swsnr** [comment]
Thank you ❤️

Would you be so kind as to push this change to npm with a new version? 😇

**Comment by schnz** [comment]
Thank you for the PR!

I had some trouble with figuring out the a working `~/.yarnrc.yml` config file. 

The new version is available now :)  (`@girs/gnome-shell@45.0.0-beta6`, or `@latest`)

**Comment by swsnr** [comment]
Thanks 🙏🏿

**Pull request merged**